### DISCUSSION TITLE
RTK Query: How to access the useMutation state in other components ?

### DISCUSSION THREAD

**Discussion by ShivamJoker** [reduxjs/redux-toolkit#1481] [Q&A]
After reading docs I can see that we can import and use the hooks and states will be synced, but if we use mutation then we also get the api's for refetching it.

So what would be the ideal way to access a read only state for the mutations ?

**Comment by phryneas** [comment]
At the moment, it is not possible. 
There is ongoing work for this in https://github.com/reduxjs/redux-toolkit/pull/1477 to enable that functionality in the next version.
Until then you'd either need to move the hook to a common parent or somehow store the result in a hand-managed slice.

**Reply by ShivamJoker** [reply]
Okay so even if we use the mutation hook it work give us the previous data which was already fetched ?

**Reply by phryneas** [reply]
Only if it was fetched by the same component. 

The point of mutations is not to fetch data, but to trigger a change on the server. Usually you are not interested in the results of this long-term, which is why they don't reside in the cache the triggering component unsubscribes or starts a new mutation.

The only example of a mutation that is an exception to this rule I can think off would be "login" or "logout" mutations which might generate something like a token - and you would have to store that to the side into a custom slice as shown in the docs.
